src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothHidDevice.cs

```diff
@@ -0,0 +1,254 @@
+        /// or when the connection attempt to the remote device fails.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void SendMouseEvent(BluetoothHidMouseData mouseData)
+        {
+            if (BluetoothAdapter.IsBluetoothEnabled && Globals.IsInitialize)
+            {
+                int ret = Interop.Bluetooth.SendHidDeviceMouseEvent(RemoteAddress, mouseData);
+                if (ret != (int)BluetoothError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to send mouse event to the remote device, Error - " + (BluetoothError)ret);
+                }
+            }
+            else
+            {
+                BluetoothErrorFactory.ThrowBluetoothException((int)BluetoothError.NotEnabled);
+            }
+        }
+
+        /// <summary>
+        /// Sends the key event data to the remote device.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown when the Bluetooth is not enabled.
+        /// or when the connection attempt to the remote device fails.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void SendKeyEvent(BluetoothHidKeyData keyData)
+        {
+            if (BluetoothAdapter.IsBluetoothEnabled && Globals.IsInitialize)
+            {
+                int ret = Interop.Bluetooth.SendHidDeviceKeyEvent(RemoteAddress, keyData);
+                if (ret != (int)BluetoothError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to send key event to the remote device, Error - " + (BluetoothError)ret);
+                }
+            }
+            else
+            {
+                BluetoothErrorFactory.ThrowBluetoothException((int)BluetoothError.NotEnabled);
+            }
+        }
+
+        /// <summary>
+        /// The DataReceived event is called when the device receives data from the HID Host.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<HidDeviceDataReceivedEventArgs> DataReceived
+        {
+            add
+            {
+                if (_hidDeviceDataReceived == null)
+                {
+                    RegisterHidDataReceivedEvent();
+                }
+                _hidDeviceDataReceived += value;
+            }
+            remove
+            {
+                _hidDeviceDataReceived -= value;
+                if (_hidDeviceDataReceived == null)
+                {
+                    UnregisterHidDataReceivedEvent();
+                }
+            }
+        }
+
+        private void RegisterHidDataReceivedEvent()
+        {
+            Interop.Bluetooth.HidDeviceDataReceivedCallback _hidDeviceDataReceivedCallback = (BluetoothHidDeviceReceivedData receivedData, IntPtr userData) =>
+            {
+                if (_hidDeviceDataReceived != null)
+                {
+                    _hidDeviceDataReceived(null, new HidDeviceDataReceivedEventArgs(receivedData));
```

Comment:
How about use null-conditional Operator?
``` c#
_hidDeviceDataReceived?.Invoke(null, new HidDeviceDataReceivedEventArgs(receivedData));
```

Reply:
Done. Thank you.